products/cloudidentity/api.yaml

```diff
@@ -0,0 +1,317 @@
+          - preferred_member_key
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'id'
+            required: true
+            description: |
+              The ID of the entity.
+
+              For Google-managed entities, the id must be the email address of an existing
+              group or user.
+
+              For external-identity-mapped entities, the id must be a string conforming
+              to the Identity Source's requirements.
+
+              Must be unique within a namespace.
+          - !ruby/object:Api::Type::String
+            name: 'namespace'
+            description: |
+              The namespace in which the entity exists.
+
+              If not specified, the EntityKey represents a Google-managed entity
+              such as a Google user or a Google Group.
+
+              If specified, the EntityKey represents an external-identity-mapped group.
+              The namespace must correspond to an identity source created in Admin Console
+              and must be in the form of `identitysources/{identity_source_id}.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'preferredMemberKey'
+        input: true
+        description: |
+          EntityKey of the member.
+        exactly_one_of:
+          - member_key
+          - preferred_member_key
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'id'
+            required: true
+            description: |
+              The ID of the entity.
+
+              For Google-managed entities, the id must be the email address of an existing
+              group or user.
+
+              For external-identity-mapped entities, the id must be a string conforming
+              to the Identity Source's requirements.
+
+              Must be unique within a namespace.
+          - !ruby/object:Api::Type::String
+            name: 'namespace'
+            description: |
+              The namespace in which the entity exists.
+
+              If not specified, the EntityKey represents a Google-managed entity
+              such as a Google user or a Google Group.
+
+              If specified, the EntityKey represents an external-identity-mapped group.
+              The namespace must correspond to an identity source created in Admin Console
+              and must be in the form of `identitysources/{identity_source_id}.
+      - !ruby/object:Api::Type::String
+        name: 'createTime'
+        output: true
+        description: |
+          The time when the Membership was created.
+      - !ruby/object:Api::Type::String
+        name: 'updateTime'
+        output: true
+        description: |
+          The time when the Membership was last updated.
+      - !ruby/object:Api::Type::Array
+        name: 'roles'
+        description: |
+          The MembershipRoles that apply to the Membership.
+          If unspecified, defaults to a single MembershipRole with name MEMBER.
```

Comment:
Is this true? It's not marked as computed. Or is it just not returned from the API?

Reply:
After double-checking, this appears to not be true? I tried not including it and got an error, so I'm just going to mark this as required.